addon/-private/system/relationships/state/relationship.js

```diff
@@ -155,19 +157,31 @@ export default class Relationship {
     heimdall.increment(addCanonicalRecord);
     if (!this.canonicalMembers.has(record)) {
       this.canonicalMembers.add(record);
-      if (this.inverseKey) {
-        record._relationships.get(this.inverseKey).addCanonicalRecord(this.record);
-      } else {
-        if (!record._implicitRelationships[this.inverseKeyForImplicit]) {
-          record._implicitRelationships[this.inverseKeyForImplicit] = new Relationship(this.store, record, this.key,  { options: {} });
-        }
-        record._implicitRelationships[this.inverseKeyForImplicit].addCanonicalRecord(this.record);
-      }
+      this.setupInverseRelationship(record, false);
     }
     this.flushCanonicalLater();
     this.setHasData(true);
   }
 
+  setupInverseRelationship(internalModel, initial) {
+    if (this.inverseKey) {
+      let relationships = internalModel._relationships;
+      let relationshipExisted = relationships.has(this.inverseKey);
+      let relationship = relationships.get(this.inverseKey);
```

Comment:
No; this PR preserves the invariant that if a relationship is initialized, its inverse is also initialized.

We could change the invariant, but it would add complexity in the case where a relationship A gets its initial state from the payload of its inverse B, B is initialized and then mutated before A is initialized.

This is something we could do but it seems better to leave that for a separate PR